mars_form_app/src/components/Form.js

```diff
@@ -0,0 +1,321 @@
+import React, {Component} from 'react';
+// import axios from 'axios';
+import countries from "./countries"
+
+class Form extends Component{
+    state = {
+        name: "",
+        birthday: "",
+        countries: countries,
+        selectedCountry: "United States",
+        diet: "",
+        explorer: "",
+        formCompleted: false,
+        formSubmitted: false,
+        breathe:"no",
+        married:false,
+        situation:"",
+        claustrophobic:false,
+        familyHistory:{},
+        living:{},
+        education:{}
+    }
+
+    // populateCountry() {
+    //     // e.preventDefault();
+    //     axios.get("http://vocab.nic.in/rest.php/country/json").then(res => {
+    //         let {countries} = res.data;
+    //         // debugger
+    //         // this.setState({countries});
+    //     })
+    // }
+    handleChange = (e) => {
+        this.setState({
+            [e.target.name]: e.target.value
+        })
+    }
+    handleSubmit = (e) => {
+        e.preventDefault();
+        this.setState({
+            formCompleted: !this.state.formCompleted
+        })
+        if(this.state.formCompleted === true) {
+            let div = document.querySelector("#formSubmitted");
+            div.hidden = false;
+        }
+    }
+
+    handleSubmitComplete = (e) => {
+        e.preventDefault();
+        let clearPage = document.querySelector("#inputPage");
+        let p = document.querySelector("#applicationClose");
+        clearPage.innerHTML = "";
+        p.hidden = false;
+    }
+    
+
+    handleCheckBox = (e,state)=>{
+        // debugger
+        if(e.target.checked){
+            let question = document.createElement("input")
+            question.type="Number"
+            question.placeholder="How Many...?"
+            e.target.parentNode.appendChild(question)
+            let submit = document.createElement("button")
+            submit.innerText="submit"
+            let inputKey = e.target.value
+            // console.log(e.target.value)
+            submit.onclick=function submitHowMany (name,value,obj){
+                name=inputKey
+                value =question.value
+                obj=state
+                obj[name]=value
+                console.log(obj)     
+            }
+            
+            e.target.parentNode.appendChild(submit)
+            let br = document.createElement("br")
+            e.target.parentNode.appendChild(br)
+        }
+    }
```

Comment:
This looks like all Unit 2 stuff. We need to write in JSX now and stop manipulating the DOM directly.